### ISSUE TITLE
Some errors after server maintenance 

### ISSUE THREAD

**Issue by mcjoshea** [Mirobit/bitcoin-node-manager#54]
After updating my server with all the latest security patches and updates, and after performing a few reboots, I checked my Bitcoin Node Manager and I'm now getting the following errors on the main page of BNM.

```
Notice: Undefined index: softforks in /.../src/Node.php on line 144
```
```
Warning: Invalid argument supplied for foreach() in /.../src/Utility.php on line 253
```

I have tried updating to the latest version of the code but that made no difference. 

I also reverted back to a snapshot taken before the updates (but not before the reboots) and I still get the same issues, so this is probably to do with the rebooting of the server, rather than anything else.  I also rebooted my Bitcoin node at a similar time, in case it matters.  I've got two separate servers.

I'm running BNM on:

> Debian Linux 10
> Linux 4.19.0-20-amd64 on x86_64
> PHP Version 7.3.31-1~deb10u1

Any ideas?

**Comment by mcjoshea** [comment]
Here are details of my node, in case it helps:

![image](https://user-images.githubusercontent.com/63942949/166335641-f0f4110b-d3f6-402b-be28-d5cd17f483fc.png)

**Comment by MeowMeNot** [comment]
This also happened to me after I updated bitcoind to version 23.  I made the edits shown below:

Go to /bitcoin-node-manager/src
---------------------------
line 144 in node.php add comment out \\
-----------------------------------------------------------------
Line 253 in Utility.php Remove:

function checkSoftFork($softForks){
	foreach($softForks as $name => &$sf){
		if($sf['type'] === 'bip9' && $sf['bip9']['status'] === "started"){
			if(!preg_match("/[A-Za-z0-9 ]{2,25}/", $name)){
				unset($softForks[$name]);
				continue;
			}
			$sf['status'] = ucfirst(preg_replace("/[^A-Za-z]/", '', $sf['bip9']['status']));
			$sf['start_time'] = date("Y-m-d",$sf['bip9']['start_time']);
			$sf['timeout'] = date("Y-m-d",$sf['bip9']['timeout']);
			$sf['since'] = checkInt($sf['bip9']['since']);
			if(isset($sf['bip9']['statistics'])){
				$sf['process'] = round(($sf['bip9']['statistics']['count']/$sf['bip9']['statistics']['period'])*100,1);
			}
		}else{
			unset($softForks[$name]);
		}
	}
	return $softForks;
}

**Comment by mcjoshea** [comment]
Thanks for that. I never even realised I had updated my node to latest version until today when I saw your message!
To get rid of the errors all I had to do was to comment out line 144 of Node.php and that did the trick.

**Comment by mcjoshea** [comment]
Just adding this note to keep my issue fresh.

**Comment by Mirobit** [comment]
This has been fixed in #60.
Thanks for reporting this.
